### PR TITLE
Add custom uri scheme for wallet

### PR DESCRIPTION
**Pull request by Qubic-Hub**
- Handle qubic-wallet://pairwc/wc:..... links to initialize wallet connect

### CODE REVIEW

**Comment by sallymoc** [comment]
@ahmed-tarek-salem could you please also ensure this is working for android?
I also saw "qubic-wallet" (the schema to be used) is included in 3 files. Would it be possible to configure in a configuration file and during building time to apply it to a xcconfig file for iOS and AndroidManifest.xml?

File: lib/pages/main/wallet_contents/add_wallet_connect/add_wallet_connect.dart [hunk 1]
```diff
@@ -254,27 +272,68 @@ class _AddWalletConnectState extends State<AddWalletConnect> {
     }
   }
 
+  String? validateWalletConnectURL(String? valueCandidate) {
+    final l10n = l10nOf(context);
```
Comment (sallymoc) [inline lib/pages/main/wallet_contents/add_wallet_connect/add_wallet_connect.dart:276]: - wallet_connect_methods.dart in lib/helpers contains similar method. Please define only one

File: lib/models/app_link/app_link_controller.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  //Handles a URL action to pair a wallet connect connection
+  void _handleWCPair(Uri uri, BuildContext context) {
+    final l10n = l10nOf(context);
+
+    String remove = "${uri.scheme}://${uri.host}/";
+    //Trim out qubic-wallet scheme and host
+    String connectionUrl = uri.toString().substring(remove.length);
+
```
Comment (sallymoc) [inline lib/models/app_link/app_link_controller.dart:37]: This is not the only validation to use. How about using the method validateWalletConnectURL here?

File: lib/models/app_link/app_link_controller.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    if (uri.pathSegments.isEmpty) {
+      throw Exception(l10n.uriInvalidFormat);
+    }
+  }
+
+  //Handles a URL action to pair a wallet connect connection
+  void _handleWCPair(Uri uri, BuildContext context) {
+    final l10n = l10nOf(context);
```
Comment (sallymoc) [inline lib/models/app_link/app_link_controller.dart:32]: This is not being used in the method

File: lib/pages/main/wallet_contents/add_wallet_connect/add_wallet_connect.dart (hunk 1 above)
Comment (ahmed-tarek-salem) [reply lib/pages/main/wallet_contents/add_wallet_connect/add_wallet_connect.dart]: It starts with "wc:", Resolved

**Pull request merged**